src/index.js

```diff
@@ -50,25 +58,40 @@ export default function (CustomElement, opts) {
       const node = ReactDOM.findDOMNode(this);
       Object.keys(props).forEach(name => {
         if (name === 'children' || name === 'style') {
-          return;
-        }
+        return;
+      }
 
-        if (name.indexOf('on') === 0 && name[2] === name[2].toUpperCase()) {
-          syncEvent(node, name.substring(2), props[name]);
-        } else {
-          node[name] = props[name];
+      if (name.indexOf('on') === 0 && name[2] === name[2].toUpperCase()) {
+        syncEvent(node, name.substring(2), props[name]);
+      } else if (name.indexOf('attr-') === 0) {
+        let attrValue = props[name];
+        if (typeof attrValue === 'object') {
+          attrValue = JSON.stringify(attrValue)
         }
-      });
+        node.setAttribute(name.substring(5), attrValue)
+      } else {
+        node[name] = props[name];
+      }
+    });
     }
     render() {
-      return React.createElement(tagName, { style: this.props.style }, this.props.children);
+      return React.createElement(
+          tagName,
+          {
+            is: this.props.is,
+            style: this.props.style
+          },
+          this.props.children
+      );
     }
   }
 
-  const proto = CustomElement.prototype;
+  const proto = typeof CustomElementOrTagName === 'string'
```

Comment:
Since there's already `CustomElementConstructor` above, could you do `CustomElementConstructor.prototype`?